tests/drivers/adc/adc_dma/boards/nucleo_l476rg.conf

```diff
@@ -0,0 +1,8 @@
+#
+# Copyright (c) 2023 Hein Wessels, Nobleo Technology
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+CONFIG_ADC_STM32_DMA=y
+CONFIG_ADC_ASYNC=y
```

Comment:
This is not strictly related to this PR but I wonder if these 2 options should stay independent.
Can we have CONFIG_ADC_ASYNC=y and CONFIG_ADC_STM32_DMA=n ? or the other way round ?

Reply:
> Can we have CONFIG_ADC_ASYNC=y and CONFIG_ADC_STM32_DMA=n ?

I believe this is possible, and that these are independent. However, the reason `CONFIG_ADC_ASYNC=y` is because there's an async test part of this suite.